### PR TITLE
ADNI-GFNet-47430004

### PR DESCRIPTION
**Pull request by DonghyugDavidJeong**
Hello COMP3710 Teaching Team.

This is my pull request to merge my ADNI-GFNet code into Shakes' topic recognition branch. The task I chose was Number 5, "Classify Alzheimer’s disease (normal and AD) of the ADNI brain data (see Appendix for link) using one of the latest vision transformers such as the GFNet set having a minimum accuracy of 0.8 on the test set."

I achieved an accuracy of 78.4% in the test set, more details are included in the README.md.

The included files are as follows: (this info is in the README.md)

`images/` - Directory that contains all the figures and images necessary for visually showing something.

`ADNI_GFNet_47430004.yaml` - yaml file which can be used to create a conda environment that is able to run the train.py. Details on using it are in [Dependencies](#dependencies).

`dataset.py` - Python file that handles all the dataset/dataloader processes.

`LICENSE` - License file containing the MIT License of the original repo found in [Inspiration](#inspiration).

~~`main.py` - Obsolete python file that was created and does not serve a purpose in the repo structure. Left as proof of progress.~~ This file has now been deleted to address a comment.

`predict.py` - Python file used to test the accuracy of a given model, which should be located at `test/model/GFNet.pth`

`README.md` - This markdown file which describes everything about the repo.

`test_dataset.py` - Python file used to visually inspect that the dataset.py file is working correctly, especially the normalising transform.

`train.py` - Python file used to train the GFNet model, and save figures relating to its accuracy over epochs, losses over epochs and the best-case accuracy.

`utils.py` - Python file containing miscellaneous functions that help the training and predict process work. This file was taken directly from the repo listed in [Inspiration](#inspiration), and the source is also listed in the header of the python file.

Thank you.
Donghyug Jeong 47430004

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.784. However, this is a validation acc as the student utilized the original train/test set to train the model and observe the validation result. The val loss graph indicates the model training tends to be overfitting.

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good. However, some discarded files and commented-off code are expected to be deleted.

**Code comment and docstring:**

- Good code comments 
- Minimal function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Fruitful and reasonable commits
- A conclusion is expected in the report to summarize the experiment.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
Pull request is descriptive about the file structure .

File Organizing: Well-organized files.

Commit Log:
Commit messages are meaningful .
Commits demonstrate a logical development flow.
Commits are regularly made.

Documentation:
The README file provides a clear explanation of the algorithm and the problem it solves.
Good idea  to have a yaml file as included . 
Code comments and docstrings should be  included .
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and  Future Works  of the report

**Comment by DonghyugDavidJeong** [comment]
Hi, is this feedback that I should be acting upon? 
I'm not exactly sure what to change and how to change it because I can't see any assigned marks anywhere, and some people have received complete comments (with marks) but some people haven't received anything.

**Comment by aniketgupta17** [comment]
They are just initial marks given by some tutors but Shakes will be providing the complete mark for the Report and Code  . Yes this is the feedback you should act upon . Incorporate the mentioned changes in your ReadMe.MD file and then push again .

**Comment by DonghyugDavidJeong** [comment]
I've tried to address the comments, but am unsure if I got all of them because I couldn't see any marks lost (usually a good indication of what to change). I hope I'll be able to make more changes later when I get more specific comments.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Model training tends to be overfitting.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure NOT present. Unwanted files present.</td><td>-1</td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. Minimal doc strings.</td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug.</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. Minimal.</td><td>-1</td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. Lacks flow.</td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-6</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by DonghyugDavidJeong** [comment]
Hi, I just wanted to ask about what "File structure NOT present. Unwanted files present." Is? Is this for the utils.py file?
